Update docs about changes that went in for kindless upgrade task (#6922)
aws/eks-anywhere: docs/content/en/docs/getting-started/vsphere/vsphere-spec.md

The VM template to use for your EKS Anywhere cluster. This template was created when you
[imported the OVA file into vSphere]({{< relref "../vsphere/customize/vsphere-ovas.md" >}}).
This is a required field if you are using Ubuntu-based or RHEL-based OVAs.

> The `template` name must have a cluster kubernetes version or workerNodeGroupConfiguration.kubernetesVersion version(in case of modular upgrade) in the name. If the kubernetes version is 1.24, the `template` field name should include 1.24, 1_24, 1-24 or 124.

Reviewer: ```suggestion
The `template` must contain the `Cluster.Spec.KubernetesVersion` or `Cluster.Spec.WorkerNodeGroupConfiguration[].KubernetesVersion` version (in case of modular upgrade). For example, if the Kubernetes version is 1.24, `template` must include 1.24, 1_24, 1-24 or 124.
```
